src/lib/eliom_registration.server.ml

```diff
@@ -107,29 +107,25 @@ module Make_typed_xml_registration
   (Typed_xml: Xml_sigs.Typed_xml with module Xml := Xml)
   (E : sig type content end) = struct
 
-    module Format = Xml_print.Make_typed(Xml)(Typed_xml)(Ocsigen_stream.StringStream)
-
-    let result_of_content_subxhtml get_etag c =
-      let encode x = fst (Xml_print.Utf8.normalize_html x) in
-      let x = Format.print_list ~encode c in
+    module Fmt = Xml_print.Make_typed_fmt(Xml)(Typed_xml)
+
+    let out l =
+      Format.asprintf "%a"
+        (Format.pp_print_list ~pp_sep:(fun _ () -> ())
+           (let encode x = fst (Xml_print.Utf8.normalize_html x) in
```

Comment:
It would probably be best to hoist that.

Also, you probably want to check the first version with pp_print_list, we may need to reimplement it locally.